src/main/java/com/microfocus/application/automation/tools/octane/model/processors/scm/StarTeamSCMProcessor.java

```diff
@@ -0,0 +1,252 @@
+ */
+
+package com.microfocus.application.automation.tools.octane.model.processors.scm;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.LinkedList;
+import java.util.List;
+import org.apache.logging.log4j.Logger;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import com.hp.octane.integrations.dto.DTOFactory;
+import com.hp.octane.integrations.dto.scm.SCMChange;
+import com.hp.octane.integrations.dto.scm.SCMCommit;
+import com.hp.octane.integrations.dto.scm.SCMData;
+import com.hp.octane.integrations.dto.scm.SCMRepository;
+import com.hp.octane.integrations.dto.scm.SCMType;
+import com.microfocus.application.automation.tools.octane.configuration.SDKBasedLoggerProvider;
+import hudson.model.AbstractBuild;
+import hudson.model.Run;
+import hudson.model.User;
+import hudson.model.UserProperty;
+import hudson.scm.ChangeLogSet;
+import hudson.scm.EditType;
+import hudson.scm.SCM;
+import hudson.scm.ChangeLogSet.Entry;
+import hudson.tasks.Mailer;
+
+class StarTeamSCMProcessor implements SCMProcessor {
+  private static final Logger logger = SDKBasedLoggerProvider.getLogger(StarTeamSCMProcessor.class);
+  private static final DTOFactory dtoFactory = DTOFactory.getInstance();
+
+  @Override
+  public SCMData getSCMData(AbstractBuild build, SCM scm) {
+    List<ChangeLogSet<? extends ChangeLogSet.Entry>> changes = new ArrayList<>();
+    changes.add(build.getChangeSet());
+    return extractSCMData(build, scm, changes);
+  }
+
+  @Override
+  public SCMData getSCMData(WorkflowRun run, SCM scm) {
+    return extractSCMData(run, scm, run.getChangeSets());
+  }
+
+  @Override
+  public CommonOriginRevision getCommonOriginRevision(final Run run) {
+    return null;
+  }
+
+  private SCMData extractSCMData(Run run, SCM scm, List<ChangeLogSet<? extends ChangeLogSet.Entry>> changes) {
+    if (!(scm.getClass().getName().equals("hudson.plugins.starteam.StarTeamSCM"))) {    
+      throw new IllegalArgumentException("SCM type of StarTeamSCM was expected here, found '" + scm.getClass().getName() + "'");
+    }
+
+    SCMRepository repository = getRepository(scm);
+    List<SCMCommit> commits = extractCommits(changes);
+
+    return dtoFactory.newDTO(SCMData.class)
+        .setRepository(repository)
+        .setCommits(commits);
+  }
+
+  private SCMRepository getRepository(SCM starTeamData) {
+    SCMRepository result = null;
+    if (starTeamData != null) {
+      try {
+        String url = getSCMRepositoryURL(starTeamData);
+        String branch = getSCMRepositoryBranch(starTeamData);
+        result = dtoFactory.newDTO(SCMRepository.class).setType(SCMType.STARTEAM).setUrl(url).setBranch(branch);
+      } catch (IllegalAccessException | IllegalArgumentException | InvocationTargetException | NoSuchMethodException | SecurityException e) {
+        logger.warn("failed to obtain BuildData; no SCM repository info will be available");
```

Comment:
please add to warn exception class name (e.getClass().getSimpleName())+ exception message